src/main/scala/com/expedia/www/haystack/commons/entities/MetricPoint.scala

```diff
@@ -28,17 +30,23 @@ import com.expedia.www.haystack.commons.entities.MetricType.MetricType
   * @param epochTimeInSeconds : epochTime in seconds for when the event is generated
   */
 case class MetricPoint(metric: String, `type`: MetricType, tags: Map[String, String], value: Float, epochTimeInSeconds: Long) {
-
   def getMetricPointKey(enablePeriodReplacement: Boolean): String = {
-   val metricTags =  if (enablePeriodReplacement) {
-      tags.foldLeft("")((tag, tuple) => {
-        tag + s"${tuple._1}.${tuple._2.replace(".", "___")}."
-      })
-    } else {
-      tags.foldLeft("")((tag, tuple) => {
-        tag + s"${tuple._1}.${tuple._2}."
-      })
-    }
+    getMetricPointKey(enablePeriodReplacement, false)
+  }
+
+  def getMetricPointKey(enablePeriodReplacement: Boolean, enableBase64Encoding: Boolean): String = {
+    val metricTags = tags.foldLeft("")((tag, tuple) => {
+      var tuple2 = tuple._2
+      if ("serviceName".equalsIgnoreCase(tuple._1) || "operationName".equalsIgnoreCase(tuple._1)) {
```

Comment:
can't we base64 encode all tag values ? we're decoding all the tag values in the createTagsFromMetricKey function